File: RouletteGame.py
# This file is an example of Tech Exam from Bruno
# Rules that need to be given to the candidate for the test :
	# The game is of "roulette" kind :
	# Player is given $1000 at the beginning of the game, and he can :
		# REQ-01: Choose the number he/she wants to deal on. As this is a roulette, numbers are given between 0 and 49.
		# REQ-02: Choose the amount he/she wants to deal,
		# REQ-03 : Play with hazard
	# The benefits are calculated with respect to the following algorithm :
		# ALG-01: If player chooses the good number, he wins 3 times his deal,
		# ALG-02: If player chose the good color, he wins 50% of his deal. The color is determined regarding if number is odd or eaven.
		# ALG-03: If not, then player looses his/her deal.
	# The end of the game occurs when :
		# REQ-04 : player doesn't have any money left,
		# REQ-05 : player chooses to leave the roulette.

import os
from random import randrange
from math import ceil
 
# Declare constants :
playerMoney = 1000
playAgain = True # Boolean : TRUE to say "play agin", FALSE otherwise
 
print("Welcome to our new Casino. You will be playing with $", playerMoney, ".")
 
while playAgain: # As long as you can continue to play
    # Ask for the number player wants to deal on
	# REQ-01: Choose the number he/she wants to deal on. As this is a roulette, numbers are given between 0 and 49.
    playerDeal = -1
    while playerDeal < 0 or playerDeal > 49:
        playerDeal = input("Please type in the number you want to deal on (between 0 and 49) : ")
        # Convert text into integer
        try:
            playerDeal = int(playerDeal)
        except ValueError:
            print("Please type in a number")
            playerDeal = -1
            continue
        if playerDeal < 0:
            print("Please type in a positive number")
        if playerDeal > 49:
            print("Please type in a number lower than 49")
 
    # Ask for the amount of money player wants to deal :
	# REQ-02: Choose the amount he/she wants to deal,
    deal = 0
    while deal <= 0 or deal > playerMoney:
        deal = input("Please type in the amount of money you want to deal on : ")
        # Convert text into integer
        try:
            deal = int(deal)
        except ValueError:
            print("Please type in a number")
            deal = -1
            continue
        if deal <= 0:
            print("Please type in a positive number")
        if deal > playerMoney:
            print("You don't have that money left... Your credit is : $", playerMoney)
 
    # Time to play !
	# REQ-03 : Play with hazard
    winNumber = randrange(50)
    print("The roulette has decided ... it stopped on number ", winNumber)
 
    # Now determine what is the benefits of player
    if winNumber == playerDeal: # ALG-01 : If player chooses the good number, he wins 3 times his deal
        print("Congratulations ! You won $", deal * 3, " !")
        playerMoney += deal * 3
    elif winNumber % 2 == playerDeal % 2: # ALG-02 : If player chose the good color, he wins 50% of his deal
        deal = ceil(deal * 0.5)
        print("You chose the good color. Your gain is $", deal)
        playerMoney += deal
    else: # ALG-03 : If not, then player looses his/her deal
        print("Sorry... you lost your deal.")
        playerMoney -= deal
 
    # If player doesn't have money left, then that's the end of the game
	# REQ-04 : player doesn't have any money left,
    if playerMoney <= 0:
        print("You lost all your money ! That's the end of the game...")
        playAgain = False
    else:
        # If not, then continue
        print("Now you have $", playerMoney)
		# Ask if player wants to continue
		# REQ-05 : player chooses to leave the roulette.
        leaveGame = input("Do you want to leave the game (y/n) ? ")
        if leaveGame == "y" or leaveGame == "Y":
            print("Bye ! See you next time !")
            playAgain = False
 
# Pause system (Windows)
os.system("pause")

File: RouletteGame.java

/**
 * Simple roulette game in Java.
 * 
 * @author Reyes, Omar Matthew B.
 * @version 
 * 2013/09/24 Converted the Python code into Java.
 * 2013/09/25 Improved error handling.
 */
import java.util.InputMismatchException;
import java.util.Random;
import java.util.Scanner;
public class RouletteGame{
    Scanner scan = new Scanner(System.in);
    // Initialize variables
    int playerMoney = 1000;
    int winNumber;
    boolean playAgain = true;
    public static void main(String [] args){        
        RouletteGame roulette = new RouletteGame();             
        roulette.playRoullete();
    }

    public void generateRandom(){
        Random random = new Random();
        //Generate random bounds with a minimum of 0 and a maximum of 49.
        winNumber = 0 + (int) (random.nextDouble() * ((49 - 0) + 1));
    }    

    public void playRoullete(){
        while(playAgain == true){
            System.out.println("Welcome to our Casino. You will be playing with $" + playerMoney +".");
            // Set the dealInput and dealPlayerMoney into -1 for them to enter
            // necessary inital looping statements.
            int dealInput =  -1, dealPlayerMoney = -1;
            // While input is not in range continue seeking appropriate input from user 
            while(dealInput < 0 || dealInput > 49){
                try{
                    System.out.println("Choose number from 0 to 49 to deal on");
                    dealInput = scan.nextInt();
                }
                //Catch InputMismatchException when input requirement is not met
                catch(InputMismatchException error){    
                    // Move to next exception.
                    scan.next();          
                }
            }

            // While input is not in range continue seeking appropriate input from user
            while(dealPlayerMoney < 0 || dealPlayerMoney > playerMoney){
                try{
                    System.out.println("Please type in the amount of money you want to deal on:");
                    dealPlayerMoney = scan.nextInt();
                }
                catch(InputMismatchException error){
                    scan.next();
                }
            }

            // Invoke generateRandom method
            generateRandom();
            System.out.println("The roullete has decided ... it stopped on number " + winNumber);

            // Roulette rule: if the input is same with the result; 
            // add dealPlayerMoney * 3
            if(winNumber == dealInput){
                System.out.println("Congratulatios! You won $" + dealPlayerMoney * 3);
                playerMoney+=dealPlayerMoney * 3;                
            }
            // Roulette rule: if the input is same with color (roulette has only two
            // colors within each block - so technically - chances are 50/50);
            // add dealPlayerMoney / 2
            else if((winNumber % 2) == (dealInput % 2)){
                dealPlayerMoney*=0.5;
                System.out.println("You chose the good color. You gain $" + dealPlayerMoney);
                playerMoney+=dealPlayerMoney;
            }
            // Roulette rule: if the input doesn't match the number and "color";
            // deduct dealPlayerMoney
            else{
                System.out.println("Sorry ... you lost your deal.");
                playerMoney-=dealPlayerMoney;
            }

            // If there's no more balance on the player's money;
            // end of game
            if(playerMoney <= 0){
                System.out.println("You lost all your money! That's the end of the game...");
                playAgain = false;
            }
            else{
                // Display current money
                System.out.println("Now you have $" + playerMoney);
                // Ask if going to play again
                System.out.println("Dou you want to leave the game (y/n)?");
                String leaveGame = scan.next();
                // Require input. Convert input into lowercase.
                if(leaveGame.toLowerCase().equals("y")){
                    System.out.println("Bye! See you next time!");
                    playAgain = false;
                    /** End Game **/
                }
                // Other than that - the game would prompt you to play again by not
                // change the boolean playAgain from being "true".
            }
        }
    }
}
